api/src/adapters/sam_gov/client.py

```diff
@@ -0,0 +1,320 @@
+    ):
+        """Initialize the client.
+
+        Args:
+            api_key: API key for SAM.gov API or a SamGovConfig object.
+            username: Username for Basic Authentication (required for sensitive data).
+            password: Password for Basic Authentication (required for sensitive data).
+            api_url: URL for the SAM.gov API. If not provided, the value from environment variable will be used.
+            extract_url: URL for the SAM.gov extract API. If not provided, the value from environment variable will be used.
+        """
+        # Extract config if provided
+        config_api_key = None
+        config_api_url = None
+        config_extract_url = None
+        if isinstance(api_key, SamGovConfig):
+            config = api_key
+            config_api_key = config.api_key
+            config_api_url = config.base_url
+            config_extract_url = config.extract_url
+            api_key = None  # Clear api_key since we've extracted it from config
+
+        # Set instance variables from parameters or environment
+        self.api_key = config_api_key or api_key or os.environ.get("SAM_GOV_API_KEY")
+        self.username = username or os.environ.get("SAM_GOV_USERNAME")
+        self.password = password or os.environ.get("SAM_GOV_PASSWORD")
+        self.api_url = config_api_url or api_url or os.environ.get("SAM_GOV_API_URL")
+        self.extract_url = (
+            config_extract_url or extract_url or os.environ.get("SAM_GOV_EXTRACT_URL")
+        )
+
+        if not self.api_url:
+            raise ValueError(
+                "API URL must be provided either directly, via config, or via environment variable."
+            )
+
+        if not self.extract_url:
+            raise ValueError(
+                "Extract URL must be provided either directly, via config, or via environment variable."
+            )
+
+        if not self.api_key:
+            raise ValueError(
+                "API key must be provided either directly or via environment variable."
+            )
+
+    def _build_headers(self, include_auth: bool = True) -> dict[str, str]:
+        """Build headers for API requests.
+
+        Args:
+            include_auth: Whether to include authentication headers.
+
+        Returns:
+            Dictionary of headers.
+        """
+        headers = {
+            "Content-Type": "application/json",
+            "Accept": "application/json",
+        }
+
+        if include_auth:
+            # Add the API key authorization
+            if self.api_key:
+                headers["x-api-key"] = self.api_key
+
+            # Add Basic Auth if username and password are provided (needed for sensitive data)
+            if self.username and self.password:
+                auth_string = f"{self.username}:{self.password}"
+                encoded_auth = base64.b64encode(auth_string.encode()).decode()
+                headers["Authorization"] = f"Basic {encoded_auth}"
```

Comment:
The entity extract endpoint takes in an api_key path parameter, it doesn't support headers or user/password as far as I see.
https://open.gsa.gov/api/sam-entity-extracts-api/